lib/api/apiUtils/authorization/prepareRequestContexts.js

```diff
@@ -118,6 +118,21 @@ function prepareRequestContexts(apiMethod, request, sourceBucket,
               generateRequestContext('objectPutACL');
             requestContexts.push(putAclRequestContext);
         }
+        // if put object with version
+        if (request.headers['x-scal-s3-version-id']) {
```

Comment:
This evaluates to `false` if the given version id is empty, but the request would still have some "special" work to do, is it intended?

Reply:
I was planning on using "null" to specify the "non-version object" version id.